pkg/registry/artifacttype/strategy_test.go

```diff
@@ -0,0 +1,263 @@
+// Copyright 2025 BWI GmbH and Artifact Conduit contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package artifacttype_test
+
+import (
+	"context"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"go.opendefense.cloud/arc/api/arc"
+	"go.opendefense.cloud/arc/pkg/registry/artifacttype"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+type testObjectTyper struct{}
+
+func (testObjectTyper) ObjectKinds(runtime.Object) ([]schema.GroupVersionKind, bool, error) {
+	return nil, false, nil
+}
+
+func (testObjectTyper) Recognizes(gvk schema.GroupVersionKind) bool {
+	return false
+}
+
+var _ = Describe("ArtifactType Strategy", func() {
```

Comment:
Strategy tests feel a bit over the top. There is a PR for validation which adds more tests, so I am inclined to leave these until strategies have been further refined.